app/lib/report_helper.rb

```diff
@@ -175,6 +184,13 @@ def resolve_params(params, data_ranges)
     new_params
   end
 
+  def all_categories
+    cat_ids = TaxonLineage.distinct.pluck(:superkingdom_taxid).join(', ')
+    taxon_name_arr = []
+    taxon_name_arr = TaxonName.connection.select_all("select taxid, name from taxon_names where taxid in (#{cat_ids})").to_hash unless cat_ids.empty?
+    taxon_name_arr
```

Comment:
What are the SQL queries that you see in your log as a result of adding this code, and how long do they take to run?

Hey @yunfangjuan these are big static tables that are only going to change when we pull in new NCBI info.   I've been thinking reading these from SQL on each request isn't really great.   Caching either in the RAILS server memory (like APC at FB) or memcache may be good for this.  Does RAILS have a caching pattern based on version increment?   I imagine when we refresh that data from NCBI, we increment a version number in SQL;  when RAILS request want to load this data, they first check the version against what they last loaded;   this saves the big query.